docs/source/cpp/streaming_execution.rst

```diff
@@ -175,9 +175,607 @@ their completion::
     // alive until this future is marked finished.
     Future<> complete = plan->finished();
 
+Constructing ``ExecNode`` using Options
+=======================================
```

Comment:
It feels like this should all go _after_ the docs about constructing an ExecPlan below, since that's where the actual code to create an exec node is introduced.